reverse-proxy.md

```diff
@@ -639,6 +639,63 @@ The examples below define the dynamic configuration in YAML files. If you rather
 
 </details>
 
+### IIS with ARR and URL Rewrite
+
+<details>
+
+<summary>click here to expand</summary>
+
+**Disclaimer:** It might be possible that the config below is not working 100% correctly, yet. Improvements to it are very welcome!
+
+#### Prerequisites
+1. **Windows Server** with IIS installed.
+2. [**Application Request Routing (ARR)**](https://www.iis.net/downloads/microsoft/application-request-routing) and [**URL Rewrite**](https://www.iis.net/downloads/microsoft/url-rewrite) modules installed.
+3. [**WebSocket Protocol**](https://learn.microsoft.com/en-us/iis/configuration/system.webserver/websocket) feature enabled.
+
+
+In the web.config example below, a Server farm named `nc-server-farm` has been created an is pointing to the Nextcloud server:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<configuration>
+  <system.web>
+    <!-- Allow all urls -->
+    <httpRuntime requestValidationMode="2.0" requestPathInvalidCharacters="" />
+  </system.web>
+  <system.webServer>
+    <rewrite>
+      <rules>
+        <!-- Force https -->
+        <rule name="Https" stopProcessing="true">
+          <match url="(.*)" />
+          <conditions>
+            <add input="{HTTPS}" pattern="^OFF$" />
+          </conditions>
+          <action type="Redirect" url="https://{HTTP_HOST}/{REQUEST_URI}" appendQueryString="false" />
+        </rule>
+        <!-- Redirect to internal nextcloud server -->
+        <rule name="To nextcloud" stopProcessing="true">
+          <match url="(.*)" />
+          <conditions>
+            <add input="{HTTPS}" pattern="^ON$" />
```

Comment:
so all queries will be forwarded to nextcloud not only if your given domain is queried?

Reply:
What hosts and certificates are used is handled elsewhere, this config is located on the site level.
I was thinking about adding a whole guide on it, but I thought if you are planing on using IIS as a reverse proxy you probably have some basic understanding of it.
FYI: https://learn.microsoft.com/en-us/iis/manage/configuring-security/how-to-set-up-ssl-on-iis
https://medium.com/@Evelyn.Taylor/how-to-setup-hostname-to-website-in-local-iis-ac3fee333526

The main point of adding this is that the normal way of creating rewrite action would look like this:
`<action type="Rewrite" url="http://nc-server-farm:11000/{R:1}" appendQueryString="false" />`
But that breaks Collabora due to how the /cool/ request is formatted, so `{UNENCODED_URL}` is required instead of the `{R:1}`

I created a post about on the forum
https://help.nextcloud.com/t/document-loading-failed-and-bad-uri-syntax-logged-in-collabora-container/200067

Reply:
> I was thinking about adding a whole guide on it 

I guess that would be nice! Feel free to create it here: https://github.com/nextcloud/all-in-one/discussions/new?category=wiki

> What hosts and certificates are used is handled elsewhere, this config is located on the site level.
I was thinking about adding a whole guide on it, but I thought if you are planing on using IIS as a reverse proxy you probably have some basic understanding of it.

I see. However is the above config then complete enough to actually make it work?

Reply:
I made the instructions a little clearer, the provided example web.config is the one I'm using, and it's working fine.

Reply:
LGTM